### PR TITLE
 fix(host, test): pin all wasmer crates to 2.0.0

### PR DESCRIPTION
**Pull request by steveeJ**
this fixes a downstream build issue that pulls in incompatible crates.
this is probably due to a wasmer bug.

fixes https://github.com/holochain/holochain/issues/1175

### CODE REVIEW

**Comment by abe-njama** [comment]
@steveeJ maybe you could add the reviewers to this PR.

File: nix/sources.nix [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+  #
+  # The fetchers. fetch_<type> fetches specs of type <type>.
+  #
+
+  fetch_file = pkgs: name: spec:
+    let
+      name' = sanitizeName name + "-src";
+    in
+      if spec.builtin or true then
```
Comment (thedavidmeister) [inline nix/sources.nix:13]: `or true`? @steveeJ

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: nix/sources.nix (hunk 1 above)
Comment (steveeJ) [reply nix/sources.nix:13]: i want to preface this by pointing out that this file is autogenerated so i didn't come up with this logic. 

consider this example:

```bash
$ nix repl
Welcome to Nix 2.5.0pre20211206_d1aaa7e. Type :? for help.

nix-repl> specWithBuiltinFalse = { builtin = false; }

nix-repl> specWithBuiltinTrue = { builtin = true; }

nix-repl> specWithoutBuiltin = { }

nix-repl>

nix-repl> specWithBuiltinFalse.builtin or true
false

nix-repl> specWithBuiltinTrue.builtin or true
true

nix-repl> specWithoutBuiltin.builtin or true
true
```

the `<expression> or <alternative expression>` syntax in Nix is not a logical OR. it's a form of handling an exception that might occur in the expression by providing an alternative expression in the exception case.

Comment (thedavidmeister) [reply nix/sources.nix:13]: @steveeJ kk

File: crates/host/src/guest.rs [hunk 2]
```diff
@@ -73,7 +73,7 @@ pub fn write_bytes(memory: &Memory, guest_ptr: GuestPtr, slice: &[u8]) -> Result
 
     // write the length prefix immediately before the slice at the guest pointer position
     for (byte, cell) in slice.iter().zip(
-        unsafe { ptr.deref_mut(memory, 0 as GuestPtr, slice.len() as Len) }
+        ptr.deref(memory, 0 as GuestPtr, slice.len() as Len)
```
Comment (steveeJ) [inline crates/host/src/guest.rs:76]: :tada:

**Review by steveeJ — COMMENTED** [review]
@freesig great to see some `unsafe` removed here :+1: i

however, the failing CI tests do make me uncomfortable. i ran the tests locally and was able to fix a few bugs. currently getting this:

```
   Compiling holochain_wasmer_host v0.0.73 (/home/steveej/src/holo/holochain-wasmer/crates/host)
error: use of deprecated associated function `holochain_wasmer_host::prelude::RuntimeError::raise`: return a Result from host functions instead
 --> src/test.rs:8:19
  |
8 |     RuntimeError::raise(Box::new(WasmError::HostShortCircuit(
  |                   ^^^^^
  |
  = note: `-D deprecated` implied by `-D warnings`
```

i'll push a commit with my changes shortly.

**Comment by steveeJ** [comment]
@freesig i reverted back to 2.0.0 but was able to fix the systematic test failure. there's still some flakiness as it seems but i'm OK with the result as it is. i think this is mergable now.

**Comment by maparent** [comment]
I know this is closed, but I'm repeating here what I said on #70 : You need 2.1.0 to compile on apple silicon.

**Comment by steveeJ** [comment]
> I know this is closed, but I'm repeating here what I said on #70 : You need 2.1.0 to compile on apple silicon.

PTAL https://github.com/holochain/holochain-wasmer/pull/72 it'll follow this one.

**Review by freesig — APPROVED** [review]
Nice one. Lets merge this then get #72 merged :)

**Comment by steveeJ** [comment]
> Nice one. Lets merge this then get #72 merged :)

please feel free to

image

:slightly_smiling_face:

**Pull request merged**